homeassistant/helpers/script.py

```diff
@@ -677,12 +677,18 @@ async def _async_call_service_step(self):
                 **params,
                 blocking=True,
                 context=self._context,
-                limit=limit,
             )
         )
         if limit is not None:
             # There is a call limit, so just wait for it to finish.
-            await service_task
+            try:
```

Comment:
This timeout disappeared and i can't tell where it went or why we did this looking at the PR and commit history. Is the lack of timeout here responsible for https://github.com/home-assistant/core/issues/98073 ? or are we expecting a timeout somewhere else?